examples/access-control-migration/src/services/casbin.authorizer.ts

```diff
@@ -0,0 +1,72 @@
+import { Provider, inject } from '@loopback/core';
+import {
+  Authorizer,
+  AuthorizationContext,
+  AuthorizationMetadata,
+  AuthorizationRequest,
+  AuthorizationDecision,
+} from '@loopback/authorization';
+import * as casbin from 'casbin';
+
+// Class level authorizer
+export class CasbinAuthorizationProvider implements Provider<Authorizer> {
+  constructor(@inject('casbin.enforcer.factory') private enforcerFactory: (name: string) => casbin.Enforcer) { }
+
+  /**
+   * @returns authenticateFn
+   */
+  value(): Authorizer {
+    return this.authorize.bind(this);
+  }
+
+  async authorize(
+    authorizationCtx: AuthorizationContext,
+    metadata: AuthorizationMetadata,
+  ): Promise<AuthorizationDecision> {
+    const subject = this.getUserName(authorizationCtx.principals[0].id);
+    const object = authorizationCtx.resourceId ?? metadata.resource ?? authorizationCtx.resource;
+
+    const request: AuthorizationRequest = {
+      subject,
+      object,
+      action: (metadata.scopes && metadata.scopes[0]) || 'execute',
```

Comment:
is `execute` defined in the casbin policy or csv files?

Reply:
`execute` is the default scope. I define `scopes` for my endpoints, but we cannot make assumption that all endpoints will be decorated with that metadata.

Reply:
Perhaps create a constant named DEFAULT_SCOPE-'execute', and use this constant instead.